tests/patmat/3144b.scala

```diff
@@ -0,0 +1,8 @@
+class Test {
+  def f(x: Any): Int = x match {
+    case xs: List[Int] @unchecked => xs.head
+    case xs: List[Int @unchecked] => xs.head
```

Comment:
You should extract this case into another function. This case is unreachable here